pyxem/data/simulated_strain.py

```diff
@@ -0,0 +1,119 @@
+import numpy as np
+from orix.quaternion import Rotation
+from diffsims.generators.simulation_generator import SimulationGenerator
+from pyxem.signals import Diffraction2D
+from pyxem.data import si_phase
+from skimage.draw import ellipse, disk
+from scipy.ndimage import gaussian_filter
+
+
+def create_diffraction_pattern(
+    simulation,
+    shape=(512, 512),
+    direct_beam_position=None,
+    radius=20,
+    num_electrons=None,
+    in_plane_angle=0,
+    calibration=0.01,
+    mirrored=False,
+    transformation_matrix=None,
+):
+    if direct_beam_position is None:
+        direct_beam_position = (shape[1] // 2, shape[0] // 2)
+    transformed = simulation._get_transformed_coordinates(
+        in_plane_angle,
+        direct_beam_position,
+        mirrored,
+        units="pixel",
+        calibration=calibration,
+    )
+    in_frame = (
+        (transformed.data[:, 0] >= 0)
+        & (transformed.data[:, 0] < shape[1])
+        & (transformed.data[:, 1] >= 0)
+        & (transformed.data[:, 1] < shape[0])
+    )
+    spot_coords = np.round(transformed.data[in_frame]).astype(int)
+    if transformation_matrix is not None:
+        direct_beam_position = direct_beam_position + (0,)
+        spot_coords = (
+            (spot_coords - direct_beam_position) @ transformation_matrix
+        ) + direct_beam_position
+    spot_intens = transformed.intensity[in_frame]
+    pattern = np.zeros(shape)
+    # checks that we have some spots
+    if spot_intens.shape[0] == 0:
+        return pattern
+    else:
+        for cord, inten in zip(spot_coords, spot_intens):
+            rr, cc = disk(cord[:2], radius, shape=shape)
+            pattern[rr, cc] = inten
+    if num_electrons is not None:
+        total = np.sum(spot_intens) * radius**2 * np.pi
+        pattern = np.random.poisson((pattern / total) * num_electrons)
+    return np.divide(pattern, np.max(pattern))
+
+
+def simulated_strain(
+    navigation_shape=(32, 32),
+    signal_shape=(512, 512),
+    disk_radius=20,
+    num_electrons=1e5,
+    strain_matrix=None,
+    lazy=False,
```

Comment:
Is it worth keeping the parameter `lazy`, since the same can be done with `simulated_strain().as_lazy()`?